pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -425,3 +437,104 @@ func (m *managerImpl) reclaimNodeLevelResources(resourceToReclaim v1.ResourceNam
+	if err != nil {
+		glog.Errorf("Could not get summary provider")
+		return false
+	}
+
+	statsFunc := cachedStatsFunc(summary.Pods)
+	evict := false
+	for _, pod := range pods {
+		podStats, ok := statsFunc(pod)
+		if !ok {
+			continue
+		}
+
+		if m.emptyDirLimitEviction(podStats, pod) {
+			evict = true
+		}
+
+		if m.containerOverlayLimitEviction(podStats, pod) {
+			evict = true
+		}
+	}
+
+	return evict
+}
+
+func (m *managerImpl) emptyDirLimitEviction(podStats statsapi.PodStats, pod *v1.Pod) bool {
+	evict := false
+	podVolumeUsed := make(map[string]*resource.Quantity)
+	for _, volume := range podStats.VolumeStats {
+		podVolumeUsed[volume.Name] = resource.NewQuantity(int64(*volume.UsedBytes), resource.BinarySI)
+	}
+	for i := range pod.Spec.Volumes {
+		source := &pod.Spec.Volumes[i].VolumeSource
+		if source.EmptyDir != nil {
+			size := source.EmptyDir.SizeLimit
+			used := podVolumeUsed[pod.Spec.Volumes[i].Name]
+			zero := resource.NewQuantity(int64(0), resource.BinarySI)
+			glog.Infof("Jing pod %s limit %v, used %v", format.Pod(pod), size, used)
+			if used != nil && size.Cmp(*zero) > 0 && used.Cmp(size) > 0 {
+				// the emptyDir usage exceeds the size limit, evict the pod
+				gracePeriod := int64(0)
+				err := m.evictPod(pod, v1.ResourceName("EmptyDir"), gracePeriod, fmt.Sprintf("emptyDir usage exceeds the limit %q", size.String()))
+				if err != nil {
+					glog.Errorf("eviction manager: pod %s exceeds emptyDir sizeLimit but failed to evict %v", format.Pod(pod), err)
+				} else {
+					glog.Infof("eviction manager: pod %s exceeds emptyDir sizeLimit and is evicted successfully", format.Pod(pod))
+					evict = true
+				}
+			}
+		}
+	}
+	return evict
+}
+
+func (m *managerImpl) containerOverlayLimitEviction(podStats statsapi.PodStats, pod *v1.Pod) bool {
+	evict := false
+	thresholdsMap := make(map[string]*resource.Quantity)
+	for _, container := range pod.Spec.Containers {
+		overlayLimit := container.Resources.Limits.StorageOverlay()
+		if overlayLimit != nil && overlayLimit.Value() != 0 {
+			thresholdsMap[container.Name] = overlayLimit
+		}
+	}
+
+	for _, containerStat := range podStats.Containers {
+		rootfs := diskUsage(containerStat.Rootfs)
+		if overlayThreshold, ok := thresholdsMap[containerStat.Name]; ok {
+			if overlayThreshold.Cmp(*rootfs) < 0 {
+				gracePeriod := int64(0)
+				err := m.evictPod(pod, v1.ResourceName("containerOverlay"), gracePeriod, fmt.Sprintf("container's overlay usage exceeds the limit %q", overlayThreshold.String()))
+				if err != nil {
+					glog.Errorf("eviction manager: pod %s exceeds container overlay limit but failed to evict %v", format.Pod(pod), err)
+				} else {
+					glog.Infof("eviction manager: pod %s exceeds container overlay limit and is evicted successfully", format.Pod(pod))
+					evict = true
+				}
+			}
+		}
+	}
+	return evict
+}
+
+func (m *managerImpl) evictPod(pod *v1.Pod, resourceName v1.ResourceName, gracePeriod int64, evictMsg string) error {
```

Comment:
It could, but adding a parameter to a function is easy enough.  Better to leave it out for now.